### PR TITLE
Create SpreadsheetLisp.ts

### PR DESCRIPTION
**Pull request by preston-johnson**
Still very much a work in progress, but I wanted to contribute a sample of what the AFE has made not only psychologically-palatable, but fun!

### CODE REVIEW

**Comment by preston-johnson** [comment]
@microsoft-github-policy-service agree [company="Living Spaces"]

**Comment by preston-johnson** [comment]
@microsoft-github-policy-service agree

**Comment by jack-williams** [comment]
Thanks for submitting this. I'll try and review this for this week, but might spill over into next week.

File: advanced-formula-environment/examples/SpreadsheetLisp.ts [hunk 1]
```diff
@@ -0,0 +1,1213 @@
...
+        "Epilogue");
+
+/*
+Table of Contents
+
+Types:
+Booleans
+Atoms
+
+Macro-dependent functions [*.xlsm]:
```
Comment (jack-williams) [inline advanced-formula-environment/examples/SpreadsheetLisp.ts:42]: Are the macro-dependent functions used by the macro functions, or is there a clean split between the two?

My preference would be to split them into into different files, so we can add a clear disclaimer for the macro-dependent ones, and then not worry about all the others.

File: advanced-formula-environment/examples/SpreadsheetLisp.ts [hunk 2]
```diff
@@ -0,0 +1,1213 @@
+/*
+
+Spreadsheet Lisp (SL) v0.7.0
+
+Forged in the Advanced Formula Environment (AFE),
+courtesy of the minds on the Excel Labs team of Microsoft Garage.
+
+All design decisions codified herein are subject to emendation,
+useful or otherwise, following sufficient argumentation.
+No rights reserved.
```
Comment (jack-williams) [inline advanced-formula-environment/examples/SpreadsheetLisp.ts:10]: I don't think this is needed because the repo has its own licence.

**Comment by jack-williams** [comment]
Thanks for creating these, lots of great functions here.

**Pull request merged**